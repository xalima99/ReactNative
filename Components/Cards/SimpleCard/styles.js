import { StyleSheet } from "react-native";
import { Cards } from "../../../constants/Colors";

export default StyleSheet.create({
  container: {
    width: 120,
    maxWidth: 120,
    height: 130,
    maxHeight: 130,
    paddingHorizontal: 16,
    paddingTop: 24,
    fontSize: 16,
    paddingBottom: 16,
    borderRadius: 8,
    alignItems: "center",
    marginVertical: 4,
  },
  imageContainer: {
    justifyContent: "center",
    alignItems: "center",
    // paddingBottom:5,
  },
  defaultText: {
    color: Cards.Simple.Default.Text,
  },
  defaultIcon: {
    tintColor: Cards.Simple.Default.Icon,
  },
  active: {
    backgroundColor: Cards.Simple.Active.BackGround,
  },
  activeText: {
    color: Cards.Simple.Active.Text,
  },
  activeIcon: {
    tintColor: Cards.Simple.Active.Icon,
  },
  icon: {
    marginBottom: 8,
    width: 56,
    height: 54,
    maxHeight: 53,
    maxWidth: 56,
    resizeMode: "center",
  },
  textContainer: {
    alignItems: "center",
    justifyContent: "center",
  },
  text: {},
});
