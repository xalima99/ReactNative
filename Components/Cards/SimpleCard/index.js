import React, { useState } from "react";
import { View, Pressable, Image, Text } from "react-native";
import PropTypes from "prop-types";
import styles from "./styles";
import { Cards } from "../../../constants/Colors";

const SimpleCard = ({ text, icon, onPress }) => {
  const [isPressed, setisPressed] = useState(false);

  return (
    <Pressable
      onPress={onPress}
      style={[
        styles.container,
        {
          backgroundColor: isPressed
            ? Cards.Simple.Active.BackGround
            : "#E7ECF2",
        },
      ]}
      onPressOut={() => setisPressed(false)}
      onPressIn={() => setisPressed(true)}
    >
      {icon && (
        <View style={[styles.imageContainer]}>
          <Image
            source={icon}
            style={[
              styles.icon,
              {
                tintColor: isPressed
                  ? styles.activeIcon.tintColor
                  : styles.defaultIcon.tintColor,
              },
            ]}
          />
        </View>
      )}
      <View style={styles.textContainer}>
        <Text
          multiline={true}
          numberOfLines={2}
          font="Regular"
          style={{
            textAlign: "center",
            color: isPressed
              ? styles.activeText.color
              : styles.defaultText.color,
            fontSize: 15,
          }}
        >
          {text}
        </Text>
      </View>
    </Pressable>
  );
};

SimpleCard.propTypes = {
  text: PropTypes.string,
  icon: PropTypes.any,
  onPress: PropTypes.func,
};

export default SimpleCard;
