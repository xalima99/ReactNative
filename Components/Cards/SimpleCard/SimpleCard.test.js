import React from "react";
import { render } from "@testing-library/react-native";
import SimpleCard from "./index";

test("should test snapshot", () => {
  const page = render(<SimpleCard />);

  expect(page.toJSON()).toMatchSnapshot();
});
