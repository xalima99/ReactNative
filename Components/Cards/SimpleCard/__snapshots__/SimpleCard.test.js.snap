// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 8,
        "fontSize": 16,
        "height": 130,
        "marginVertical": 4,
        "maxHeight": 130,
        "maxWidth": 120,
        "paddingBottom": 16,
        "paddingHorizontal": 16,
        "paddingTop": 24,
        "width": 120,
      },
      Object {
        "backgroundColor": "#E7ECF2",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      font="Regular"
      multiline={true}
      numberOfLines={2}
      style={
        Object {
          "color": "#333",
          "fontSize": 15,
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
